### PR TITLE
fix error parsing when running on directory

### LINKED ISSUE #530: How to execute doctests of a directory-tree?
First, thank's for your project!! You did a  gatefull work.

I have a Question, a CahngeRequest or properbly a bug.

I whant to start a doctest of all Notebooks within a folder (recursivly)

Example: `!nbqa doctest /home/jovyan/work/mandanten/demo/DocTest/ToTest`   

But then a got a crasy stacktrace:

```
Traceback (most recent call last):
  File "/opt/conda/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/conda/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/opt/conda/lib/python3.8/doctest.py", line 2793, in <module>
    sys.exit(_test())
  File "/opt/conda/lib/python3.8/doctest.py", line 2785, in _test
    failures, _ = testfile(filename, module_relative=False,
  File "/opt/conda/lib/python3.8/doctest.py", line 2055, in testfile
    text, filename = _load_testfile(filename, package, module_relative,
  File "/opt/conda/lib/python3.8/doctest.py", line 232, in _load_testfile
    with open(filename, encoding=encoding) as f:
IsADirectoryError: [Errno 21] Is a directory: '/tmp/tmp4eum5sau/home/jovyan/work/mandanten/demo/DocTest/ToTest'
```
Why is it searching in a temp folder?

Thanks in advise
andy

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #530

### CODE REVIEW

File: tests/tools/test_doctest.py [hunk 1]
```diff
@@ -77,3 +78,23 @@ def test_doctest_invalid_import(capsys: "CaptureFixture") -> None:
...
     _, err = capsys.readouterr()
     assert "ModuleNotFoundError: No module named 'thisdoesnotexist'" in err
+
+
+@pytest.mark.skipif(
+    sys.platform.startswith("win"), reason="Permission denied during CI"
+)
+def test_doctest_on_directory(capsys: "CaptureFixture") -> None:
+    """
+    Check that correct error is reported if notebook contains unimportable imports.
```
Comment (girip11) [inline tests/tools/test_doctest.py:88]: you might want to fix the docstring for this test. Rest of the changes looks good.

Comment (MarcoGorelli) [reply tests/tools/test_doctest.py:88]: :rofl: oops, thanks!

Comment (MarcoGorelli) [inline tests/tools/test_doctest.py:88]:
```suggestion
    Check that IsADirectoryError doesn't report temporary folder
```

File: tests/tools/test_doctest.py [hunk 2]
```diff
@@ -77,3 +78,23 @@ def test_doctest_invalid_import(capsys: "CaptureFixture") -> None:
...
     _, err = capsys.readouterr()
     assert "ModuleNotFoundError: No module named 'thisdoesnotexist'" in err
+
+
+@pytest.mark.skipif(
+    sys.platform.startswith("win"), reason="Permission denied during CI"
+)
+def test_doctest_on_directory(capsys: "CaptureFixture") -> None:
+    """
+    Check that IsADirectoryError doesn't report temporary folder
```
Comment (MarcoGorelli) [inline tests/tools/test_doctest.py:88]:
```suggestion
    Check that IsADirectoryError doesn't report temporary folder.
```

**Pull request merged**